jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java

```diff
@@ -593,7 +588,180 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
+   * @return The response by deserialize as the T
+   * @throws IOException if an exception is thrown
+   * @throws InterruptedException if an exception is thrown
+   */
+  public <T> T delete(String requestUri, MediaType expectedMediaType, Type type, HttpStatus expectedStatus)
+      throws IOException, InterruptedException {
+    HttpResponse<String> response = this.deleteContent(requestUri, expectedMediaType, expectedStatus);
+    return WebServiceUtilities.deserializeResponse(response, expectedMediaType, type);
+  }
+
+  /**
+   * Execute a web service delete.
+   * @param requestUri The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param expectSuccess Assert a success code was returned
+   * @return The response as a string
+   * @throws IOException if an exception is thrown
+   * @throws InterruptedException if an exception is thrown
+   */
+  public HttpResponse<String> delete(String requestUri, MediaType expectedMediaType, boolean expectSuccess)
+      throws IOException, InterruptedException {
+    return this.deleteContent(requestUri, expectedMediaType, expectSuccess);
+  }
+
+  /**
+   * Execute a web service delete.
+   * @param requestUri The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param expectedStatus Assert a specific status code was returned
+   * @return The response as a string
+   * @throws IOException if an exception is thrown
+   * @throws InterruptedException if an exception is thrown
+   */
+  public HttpResponse<String> delete(String requestUri, MediaType expectedMediaType, HttpStatus expectedStatus)
+      throws IOException, InterruptedException {
+    return deleteContent(requestUri, expectedMediaType, expectedStatus);
+  }
+
+  /**
+   * Do a web service delete for the given uri.
+   * @param requestUri The request uri
+   * @param returnMediaType The expected response media type
+   * @param header custom header to be set
+   * @param expectSuccess Assert a success code was returned
+   * @return A http response message
+   * @throws IOException if an exception is thrown
+   * @throws InterruptedException if an exception is thrown
+   */
+  public HttpResponse<String> delete(String requestUri, MediaType returnMediaType,
+      Map<String, String> header, boolean expectSuccess) throws IOException, InterruptedException {
+    return deleteContent(requestUri, returnMediaType, header, expectSuccess);
+  }
+
+  /**
+   * Do a web service delete for the given uri.
+   * @param requestUri The request uri
+   * @param returnMediaType The expected response media type
+   * @param header custom header to be set
+   * @param expectedStatus Assert a success code was returned
+   * @return A http response message
+   * @throws IOException if an exception is thrown
+   * @throws InterruptedException if an exception is thrown
+   */
+  public HttpResponse<String> delete(String requestUri, MediaType returnMediaType,
```

Comment:
What is the reason for passing in a Map for a header?

Reply:
got rid of the header for now. Will talk with team about it for future work